### PR TITLE
Describe service freezing [ECR-4227]

### PR DESCRIPTION
**Pull request by slowli**
This PR describes service freezing in general docs.

---

See: https://jira.bf.local/browse/ECR-4227

### CODE REVIEW

File: src/architecture/service-lifecycle.md [hunk 1]
```diff
@@ -64,20 +64,38 @@ may diverge on different nodes of the network.
...
+  to the core.
 
 The core logic is responsible for persisting artifacts and services
 across node restarts.
 
+A **stopped** service no longer participates in business logic, i.e.,
+it does not process transactions or hooks, and does not interact with the users
+in any way. Service data becomes unavailable for the other services,
+but still exists. The service name and identifier remain reserved
+for the stopped service and can't be used again for adding new services.
```
Comment (dmitry-timofeev) [inline src/architecture/service-lifecycle.md]: I am not familiar with the docs on resuming, but would it make sense saying that a stopped service may be resumed (with the same name)?

File: src/architecture/service-lifecycle.md [hunk 2]
```diff
@@ -64,20 +64,38 @@ may diverge on different nodes of the network.
...
+
+**Frozen** service state is similar to the stopped one, except the service
+state can be read both by internal readers (other services) and external ones
+(HTTP API handlers).
+
+The transitions among possible service states (including data migrations
+we discuss [below](#data-migrations)) are as follows:
+
+![Service transitions](../images/service-states.png)
```
Comment (dmitry-timofeev) [inline src/architecture/service-lifecycle.md]: Will it be added separately?

Comment (slowli) [reply src/architecture/service-lifecycle.md]: ? The file is present in this PR.

File: src/architecture/service-lifecycle.md (hunk 1 above)
Comment (slowli) [reply src/architecture/service-lifecycle.md]: Yeah, good idea 👍

File: src/architecture/service-lifecycle.md (hunk 2 above)
Comment (dmitry-timofeev) [reply src/architecture/service-lifecycle.md]: I see, sorry.

File: src/architecture/service-lifecycle.md (hunk 1 above)
Comment (slowli) [reply src/architecture/service-lifecycle.md]: Done in 3b27163.

**Pull request merged**